consensus/safety-rules/src/safety_rules.rs

```diff
@@ -462,6 +466,41 @@ impl SafetyRules {
         let signature = self.sign(timeout)?;
         Ok(signature)
     }
+
+    fn guarded_sign_commit_vote(
+        &mut self,
+        ledger_info: LedgerInfoWithSignatures,
+        new_ledger_info: LedgerInfo,
+        verifier: &ValidatorVerifier,
+    ) -> Result<Ed25519Signature, Error> {
+        self.signer()?;
+
+        // Verify that ledger_info contains at least 2f + 1 dostinct signatures
+        match verifier
+            .verify_aggregated_struct_signature(ledger_info.ledger_info(), ledger_info.signatures())
+        {
+            Ok(_) => {}
+            Err(error) => {
+                // TODO: Panic here
+                return Err(Error::InvalidQuorumCertificate(error.to_string()));
+            }
+        };
+
+        let old_ledger_info = ledger_info.ledger_info();
+
+        if !old_ledger_info
+            .commit_info()
+            .match_info(new_ledger_info.commit_info())
+        {
+            return Err(Error::InconsistentExecutionResult);
```

Comment:
probably should be explicit about which fields don't match or just include two block info's fields

Reply:
Great idea